source/core/shape.cc

```diff
@@ -104,6 +124,137 @@ Shape<dim>::align_and_center(const Point<dim> &evaluation_point) const
   return translated_point;
 }
 
+template <int dim>
+Point<dim>
+Shape<dim>::reverse_align_and_center(const Point<dim> &evaluation_point) const
+{
+  // Translation and rotation to standard position and orientation for
+  // distance calculations
+  Point<dim> center_of_rotation = position;
+  Point<dim> rotated_point;
+  Point<dim> translated_point;
+
+  // Rotation from the solid orientation
+  // Angular position around x, y and z axis
+  Tensor<1, 3> theta = orientation;
+
+  // The centralized point is the one to be rotated, and it is updated after
+  // each rotation around one axis The centralized rotated point is the result
+  // of each rotation, and it is initialized in case no rotation is performed
```

Comment:
```suggestion
  // The centralized point is the one to be rotated, and it is updated after
  // each rotation around one axis. The centralized rotated point is the result
  // of each rotation, and it is initialized in case no rotation is performed
```